fixtures/duration_type_test/test/duration_type_test.dart

```diff
@@ -3,22 +3,7 @@ import '../duration_type_test.dart';
 
 void main() {
   final api = Api.load();
-  test('addDurationSeconds', () {
-    expect(api.addDurationSeconds(2, 2), 4);
-  });
-  test('addDurationMilliseconds', () {
-    expect(api.addDurationMilliseconds(2000, 2000), 4000);
-  });
-  test('addDurationMicroseconds', () {
-    expect(api.addDurationMicroseconds(2000000, 2000000), 4000000);
-  });
-  test('getBackDurationSeconds', () {
-    expect(api.getBackDurationSeconds(4), 4);
-  });
-  test('getBackDurationMilliseconds', () {
-    expect(api.getBackDurationMilliseconds(4000), 4000);
-  });
-  test('getBackDurationMicroseconds', () {
-    expect(api.getBackDurationMicroseconds(4000000), 4000000);
-  });
+  final duration = api.addDuration(1, 1);
+  expect(api.getSeconds(duration), equals(1));
+  expect(api.getNanos(duration), equals(1));
```

Comment:
maybe we can use some more complicated values than just a `1`?

Reply:
Yeah I'll add more test cases, the PR is not yet finalised. I am beginning to see actual errors in type generator now...